src/Mailgun/HttpClientConfigurator.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace Mailgun;
+
+use Http\Client\HttpClient;
+use Http\Client\Common\PluginClient;
+use Http\Discovery\HttpClientDiscovery;
+use Http\Discovery\UriFactoryDiscovery;
+use Http\Message\UriFactory;
+use Http\Client\Common\Plugin;
+
+/**
+ * Configure a HTTP client.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class HttpClientConfigurator
+{
```

Comment:
Are we happy with this client configurator? @sagikazarmark was it this you had in mind?